connector.rb

```diff
@@ -1,37 +1,2798 @@
+              optional: false,
+              control_type: :password,
+              hint: 'The 4me bearer token'
+            }
+          ],
+
+          apply: lambda do |connection|
+            if current_url.include?('https://graphql.')
+              bearer_token = connection['bearer_token']
+              headers(Authorization: "Bearer #{bearer_token}") unless bearer_token.blank?
+            end
+          end
+        },
+
+        oauth2_client_credentials: {
+          type: 'custom_auth',
+
+          fields: [
+            {
+              name: 'client_id',
+              label: 'Client ID',
+              optional: false,
+              hint: 'The 4me OAuth 2.0 client ID'
+            },
+            {
+              name: 'client_secret',
+              label: 'Client secret',
+              optional: false,
+              control_type: :password,
+              hint: 'The 4me OAuth 2.0 client secret'
+            }
+          ],
+
+          acquire: lambda do |connection|
+            instance = connection['instance']
+            region = connection['region']
+            custom_domain_name = connection['custom_domain_name']
+
+            token_url =
+              case instance
+              when 'production'
+                case region
+                when 'au'
+                  'https://oauth.au.4me.com/token'
+                when 'uk'
+                  'https://oauth.uk.4me.com/token'
+                when 'us'
+                  'https://oauth.us.4me.com/token'
+                when 'ch'
+                  'https://oauth.ch.4me.com/token'
+                else
+                  'https://oauth.4me.com/token'
+                end
+              when 'quality_assurance'
+                case region
+                when 'au'
+                  'https://oauth.au.4me.qa/token'
+                when 'uk'
+                  'https://oauth.uk.4me.qa/token'
+                when 'us'
+                  'https://oauth.us.4me.qa/token'
+                when 'ch'
+                  'https://oauth.ch.4me.qa/token'
+                else
+                  'https://oauth.4me.qa/token'
+                end
+              when 'demo'
+                'https://oauth.4me-demo.com/token'
+              else
+                "https://oauth.#{custom_domain_name}/token"
+              end
+
+            request = post(token_url)
+            request.headers('x-4me-Account': connection['account'])
+            payload = {
+              grant_type: 'client_credentials'
+            }.compact
+            payload = payload.merge({
+              client_id: connection['client_id'],
+              client_secret: connection['client_secret']
+            })
+            request = request.payload(payload)
+            request.request_format_www_form_urlencoded
+          end,
+
+          apply: lambda do |connection|
+            if current_url.include?('https://graphql.')
+              token_type = connection['token_type'] || 'Bearer'
```

Comment:
Did we forget to define `token_type` or remove it here?

Reply:
It can be removed, the token type is always `Bearer`